lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,2106 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+
+module: cloudfront_distribution
+
+short_description: create, update and delete aws cloudfront distributions.
+
+description:
+    - Allows for easy creation, updating and deletion of CloudFront
+      distributions.
+
+requirements:
+  - boto3 >= 1.0.0
+  - python >= 2.6
+
+version_added: "2.4"
+
+author: Willem van Ketwich (@wilvk)
+
+options:
+
+    state:
+      description:
+        - The desired state of the distribution or streaming distribution.
+          present - creates a new distribution or updates an existing
+            distribution.
+          absent - deletes an existing distribution.
+      choices: ['present', 'absent']
+      required: false
+
+    distribution_id:
+      description:
+        - The id of the cloudfront distribution.
+          This parameter can be exchanged with C(alias) or C(caller_reference)
+          and is used in conjunction with C(e_tag).
+          Used with C(streaming_distribution=no)
+      required: false
+
+    streaming_distribution_id:
+      description:
+        - The id of the CloudFront streaming distribution.
+          This parameter can be exchanged with C(alias) and is used in
+          conjunction with C(e_tag).
+          Used with C(streaming_distribution=yes).
+      required: false
+
+    e_tag:
+      description:
+        - A unique identifier of a modified or existing distribution.
+          Used in conjunction with C(distribution_id) or
+          C(streaming_distribution_id).
+          Is determined automatically if not specified.
+      required: false
+
+    generate_presigned_url:
+      description:
+        - Generates a presigned url for a distribution from a private
+          I(.pem) file key.
```

Comment:
Maybe C(.pem)? I'm not sure you need to specify that .pem is a file type since it's just the extension. @gundalow ?